Documentation/prestotables.md

```diff
@@ -2,25 +2,62 @@
-- `createTableAs`: Optional: If true, then `query` must also be set. When `createTableAs` is true it causes reporting-operator to create a table within Presto using `query` as the SELECT statement for creating the table, using the schema of the query and the results as the content of the table. `columns` must still be set in order for other Resources to determine the correct schema for the table, despite the schema being determined from the query used.
-- `query`: Optional: If non-empty, you must set `view` or `createTableAs` to true. Must be a SELECT query to be used for creating the table or view.
+For more information on the specific fields listed below, follow the respective links to the Presto concepts documentation.
+##### Required Fields:
+- `unmanaged`: Indicates whether a PrestoTable resource is referencing an existing table, and if set to true, the reporting-operator should not attempt to create or manage the table within Presto.
+- `catalog`: The [catalog](https://prestosql.io/docs/current/overview/concepts.html#catalog) the Presto table is created within, or the catalog the table should exist within if unmanaged. In many cases, this will be `hive`.
+- `schema`: The [schema](https://prestosql.io/docs/current/overview/concepts.html#schema) within the Presto catalog for the table to created in, or the schema the table should exist in if unmanaged. If the catalog is `hive` then there will always be at least the `default` schema.
+- `tableName`: The desired name of the [table](https://prestosql.io/docs/current/overview/concepts.html#table) to be created in Presto, or the name of an existing table if unmanaged.
+- `columns`: A list of columns that match the schema of the PrestoTable. For each list item, you must specify a `name` field, which is the name of an individual column for the Presto table, and a `type` field, which corresponds to a valid [Presto type.](https://prestosql.io/docs/current/language/types.html)
+##### Optional Fields:
+- `query`: The SELECT [query](https://prestosql.io/docs/current/overview/concepts.html#query) used for creating the table or view. If `query` is non-empty, you must set either `view` or `createTableAs` to true. Continue onto the next section for [examples](#example-prestotables) that use the `query` field.
+- `view`: Controls whether the reporting-operator needs to create a view within Presto. If true, the reporting-operator uses the `query` field as the SELECT statement used to create the table, using both the schemas of the query and results for the content of the table.
+- `createTableAs`: Controls whether the reporting-operator needs to create a table within Presto using the `query` fields as the SELECT statement for creating the table.
+- `properties`: A map containing string keys and values, where each key-value pair is a table property for configuring the table. The available properties depend on the `catalog` specified. See the Presto documentation for information on available properties.
```

Comment:
@chancez  "See the Presto documentation for information on available properties."
I wasn't able to find this when I ran a quick google search.

Reply:
It depends on the catalog, so it's per connector. We could potentially just say something to that end. "See the Presto connector documentation to find available properties for your connector's catalog/."

Reply:
:+1: